staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go

```diff
@@ -53,7 +53,9 @@ type AnnotateOptions struct {
 	// Common user flags
 	overwrite       bool
 	local           bool
-	dryrun          bool
+	dryRunClient    bool
+	dryRunServer    bool
```

Comment:
these two booleans have 4 states, one of which is invalid `(true, true)`. you could use the `DryRunStrategy` directly in there?